x/distribution/keeper/key.go

```diff
@@ -1,55 +1,136 @@
 package keeper
 
 import (
+	"encoding/binary"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-// keys/key-prefixes
+const (
+	// default paramspace for params keeper
+	DefaultParamspace = "distr"
+)
+
+// keys
 var (
-	FeePoolKey               = []byte{0x00} // key for global distribution state
-	ValidatorDistInfoKey     = []byte{0x01} // prefix for each key to a validator distribution
-	DelegationDistInfoKey    = []byte{0x02} // prefix for each key to a delegation distribution
-	DelegatorWithdrawInfoKey = []byte{0x03} // prefix for each key to a delegator withdraw info
-	ProposerKey              = []byte{0x04} // key for storing the proposer operator address
+	FeePoolKey            = []byte{0x00} // key for global distribution state
+	ProposerKey           = []byte{0x01} // key for the proposer operator address
+	OutstandingRewardsKey = []byte{0x02} // key for outstanding rewards
+
+	DelegatorWithdrawAddrPrefix          = []byte{0x03} // key for delegator withdraw address
+	DelegatorStartingInfoPrefix          = []byte{0x04} // key for delegator starting info
+	ValidatorHistoricalRewardsPrefix     = []byte{0x05} // key for historical validators rewards / stake
+	ValidatorCurrentRewardsPrefix        = []byte{0x06} // key for current validator rewards
+	ValidatorAccumulatedCommissionPrefix = []byte{0x07} // key for accumulated validator commission
```

Comment:
It would be ideal for this to be in the same validator struct as staking, but thats not possible atm due to no column-hashing / cross-module leaf combining support.

Reply:
Sure, in the future.